packages/web/src/components/RedwoodProvider.js

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'react'
+
+import { GraphQLProvider, createGraphQLClient } from 'src/graphql'
+
+const useAuthStub = () => ({ loading: false, authenticated: false })
+
+/**
+ * Redwood's Provider ties together our authentication and Apollo's GraphQL Provider
+ * when you pass the `useAuth` hook from `@redwoodjs/auth`
+ * The authentication token is added to the headers of the GraphQL client.
+ */
+const RedwoodProvider = ({ useAuth = useAuthStub, children }) => {
+  const { loading, authenticated, getToken, type } = useAuth()
+  const [authToken, setAuthToken] = useState()
+
+  useEffect(() => {
+    if (authenticated) {
+      getToken().then((token) => setAuthToken(token))
+    }
+  }, [authenticated, getToken])
+
+  // This really sucks because rendering is completely blocked whilst we're
+  // restoring authentication.
+  if (loading) {
+    return null
+  }
+  // If we have an authToken then modify the headers of the GraphQL client.
+  const client = authToken
+    ? createGraphQLClient({
+        headers: {
+          // Used by `api` to determine the auth type.
+          'X-Redwood-Auth-Type': type,
```

Comment:
Would be nice if we had shared constants across web and api

Reply:
That's a great point, I couldn't think of a nice way of doing that without making them dependencies of one another. I could share them via a 3rd party package, or via `@redwoodjs/internal`, but then I have to inject it into webpack.

It just seemed like way more added complexity than making a note, and felt like it would decrease the stability rather than improve it.

Reply:
Right on! Totally agreed. Something to figure out if it becomes a pain in the future.